pdsc/localization.py

```diff
@@ -726,6 +743,64 @@ def __init__(self, metadata):
             corners, 1.0, 1.0, 1
         )
 
+class LrocCdrLocalizer(FourCornerLocalizer):
+    """
+    A localizer for the LROC CDR  observations (subclass of
+    :py:class:`FourCornerLocalizer`)
+    """
+
+    DEFAULT_RESOLUTION_M = 1/3*(1e-6)
+    """
+    Sets the default resolution for lroc CDR localization
+    1/3 of that of HiRISE
+    """
+
+    NORMALIZED_PIXEL_SPACE = False
+
+    def __init__(self, metadata):
+        """
+        :param metadata:
+            "lroc_cdr" :py:class:`~pdsc.metadata.PdsMetadata` object
+        """
+        corners = np.array([
+            [metadata.upper_left_latitude, metadata.upper_left_longitude],
+            [metadata.upper_right_latitude, metadata.upper_right_longitude],
+            [metadata.lower_right_latitude, metadata.lower_right_longitude],
+            [metadata.lower_left_latitude, metadata.lower_left_longitude],
+        ])
+        super(LrocCdrLocalizer, self).__init__(
+            corners, metadata.lines, metadata.samples, 1
+        )
+
+class LrocCdrBrowseLocalizer(FourCornerLocalizer):
```

Comment:
Should `LrocCdrBrowseLocalizer` class be a subclass of `LrocCdrLocalizer` instead of `FourCornerLocalizer`?

Reply:
Hi Steven, I'm not sure how to change this... when I make LrocCdrBrowseLocalizer a subclass of LrocCdrLocalizer, I get that I give it too many inputs to init.
I looked at the HiRISE functions, and HiRiseBrowse doesn't use the FourCorner class. The HiRISE class that uses the FourCorner localizer is the HiRiseNoMapLocalizer, but there is no browse version of this.... so, I'm not really sure how to do this. Do you know how I can update?

I've made all the other updates though! Thank you so much for both your help! :) And, I think one of my first ever code reviews!

Reply:
Hi again Steven, I figured out how to make LrocCdrBrowseLocalizer inherit from LrocCdrLocalizer and pushed my changes to the LROC branch! The one thing I had to do though was make LrocCdrLocalizer have a downsample parameter, but I initialize that to one when I create the lroc_cdr localizer:[(https://github.com/JPLMLIA/pdsc/blob/lroc/pdsc/localization.py#L909) Is this ok, or is there a different way to do this?  

Also, I added unit tests for alternative downsample amounts for browse imagery (other than the standard 2.0), and put those here:[ ](https://github.com/JPLMLIA/pdsc/blob/lroc/test/test_localization.py#L486)